frontend/src/scenes/authentication/loginLogic.ts

```diff
@@ -57,39 +65,48 @@ export const loginLogic = kea<loginLogicType>({
                 },
             },
         ],
-        authenticateResponse: [
-            null as AuthenticateResponseType | null,
-            {
-                authenticate: async ({ email, password }: { email: string; password: string }) => {
-                    try {
-                        await api.create('api/login', { email, password })
-                        return { success: true }
-                    } catch (e) {
-                        return {
-                            success: false,
-                            errorCode: (e as Record<string, any>).code,
-                            errorDetail: (e as Record<string, any>).detail,
-                        }
-                    }
-                },
+    })),
+
+    forms(({ actions }) => ({
+        login: {
+            defaults: {} as unknown as LoginForm,
+            errors: ({ email, password }) => ({
+                email: !email ? 'Please enter your email to continue' : undefined,
+                password: !password
+                    ? 'Please enter your password to continue'
+                    : password.length < 8
+                    ? 'Password must be at least 8 characters'
+                    : undefined,
+            }),
+            submit: async ({ email, password }, breakpoint) => {
+                await breakpoint()
+                try {
+                    return await api.create('api/login', { email, password })
+                } catch (e) {
+                    actions.setLoginManualErrors({
+                        generic: {
+                            code: (e as Record<string, any>).code,
+                            detail: (e as Record<string, any>).detail,
+                        },
+                    })
+                    throw e
+                }
             },
-        ],
-    }),
-    listeners: {
-        authenticateSuccess: ({ authenticateResponse }) => {
-            if (authenticateResponse?.success) {
-                handleLoginRedirect()
-                // Reload the page after login to ensure POSTHOG_APP_CONTEXT is set correctly.
-                window.location.reload()
-            }
         },
-    },
-    urlToAction: ({ actions }) => ({
+    })),
+    listeners({
+        submitLoginSuccess: () => {
+            handleLoginRedirect()
+            // Reload the page after login to ensure POSTHOG_APP_CONTEXT is set correctly.
+            window.location.reload()
```

Comment:
This is an interesting case. We use `replaceState` to change the `login` URL, and then reload the browser. Clever, if the goal is to remove the back button... 🤔

Reply:
Yeah copy-pasted from the previous solution so someone else's clever solution ;)